riscv/processor.cc

```diff
@@ -724,8 +745,10 @@ void processor_t::set_privilege(reg_t prv, bool virt)
   mmu->flush_tlb();
   state.prev_prv = state.prv;
   state.prev_v = state.v;
+  state.prv_changed = true;
   state.prv = legalize_privilege(prv);
   state.v = virt && state.prv != PRV_M;
+  state.v_changed = state.v != state.prev_v;
```

Comment:
This probably isn't a bug, but I thought it was worth asking about.  `prv_changed` is unconditionally set to `true` here, even though it might not have actually changed.  (Consider: MRET returning from M-mode to M-mode.)  However, we are being precise about `v_changed`.

I feel like we should be consistent here.  Is there any harm in setting `v_changed` more sloppily?  Alternatively, is there any harm in setting `prv_changed` more precisely?

Reply:
@aswaterman --good suggestion -- will post an update. BTW, this PR has some `encoding.h` artifacts from the approved, but not yet merged https://github.com/riscv-software-src/riscv-isa-sim/pull/1383, so if you planning on committing that soon, I can post a revised version of that as well.

Reply:
Merged #1383.

Reply:
Thanks @aswaterman. Please see if the update fits the bill.

Reply:
LGTM.

Reply:
Great -- I will push an update  for the `encoding.h ` commit when https://github.com/riscv/riscv-opcodes/pull/181 gets merged.

Reply:
Merged that, too.  Sorry for the long dependence chain... :)